x-pack/plugins/apm/server/lib/fleet/register_fleet_policy_callbacks.ts

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { APMPlugin, APMRouteHandlerResources } from '../..';
+import { listConfigurations } from '../settings/agent_configuration/list_configurations';
+import { setupRequest } from '../helpers/setup_request';
+import { APMPluginStartDependencies } from '../../types';
+import { ExternalCallback } from '../../../../fleet/server';
+import { AGENT_NAME } from '../../../common/elasticsearch_fieldnames';
+import { AgentConfiguration } from '../../../common/agent_configuration/configuration_types';
+
+export async function registerFleetPolicyCallbacks({
+  plugins,
+  ruleDataClient,
+  config,
+  logger,
+}: {
+  plugins: APMRouteHandlerResources['plugins'];
+  ruleDataClient: APMRouteHandlerResources['ruleDataClient'];
+  config: NonNullable<APMPlugin['currentConfig']>;
+  logger: NonNullable<APMPlugin['logger']>;
+}) {
+  if (!plugins.fleet) {
+    return;
+  }
+  const fleetPluginStart = await plugins.fleet.start();
+
+  registerAgentConfigExternalCallback({
+    fleetPluginStart,
+    callbackName: 'packagePolicyCreate',
```

Comment:
I wonder if it is really necessary to have this callback `packagePolicyCreate`? When a policy is created it doesn't have an APM integration, so nothing is gonna happen, am I right? On the other hand, when you add APM integration to a policy it probably calls the `packagePolicyUpdate`.

Reply:
I made some tests and it looks like it's the other way around the `packagePolicyCreate` is called when APM integration is added to a policy. Hence I'm not sure when `packagePolicyUpdate` is invoked, would you know?

Reply:
I'm not sure exactly when it would be implicitly invoked, but it could be explicitly invoked at any time outside of the APM plugin, so i think it's OK to keep it defined just in case.